### PR TITLE
Cross publish to both Scala 2.12 and 2.13

### LINKED ISSUE #115: Support Scala 2.12 & 2.13
We should support both two latest scala versions by enabling cross-compiling:

```scala
ThisBuild / crossScalaVersions := Seq("2.12.10", "2.13.1")
```

### PR DESCRIPTION
**Pull request by alejandrohdezma**
# What has been done in this PR?

- Simplify project hierarchy
- Make necessary changes in order to support Scala 2.13
- Add cross-compilation for both Scala `2.12.10` and `2.13.1`

# References issues

This PR closes #115

### CODE REVIEW

File: memeid4s-http4s/src/main/scala/memeid4s/http4s/instances.scala [hunk 1]
```diff
@@ -36,7 +34,7 @@ trait instances {
 
   /** Allow using UUIDs in a request's query params */
   implicit val UUIDQueryParamEncoderInstance: QueryParamEncoder[UUID] =
-    QueryParamEncoder[String].contramap(_.show)
+    QueryParamEncoder[String].contramap(_.toString)
```
Comment (BenFradet) [inline memeid4s-http4s/src/main/scala/memeid4s/http4s/instances.scala]: out of curiosity, why was this change necessary?

Comment (alejandrohdezma) [reply memeid4s-http4s/src/main/scala/memeid4s/http4s/instances.scala]: To be able to not depend on `memeid-cats`

Comment (BenFradet) [reply memeid4s-http4s/src/main/scala/memeid4s/http4s/instances.scala]: ah makes sense, thanks :+1:
